src/functions/template-tags/general.php

```diff
@@ -1380,8 +1380,11 @@ function tribe_events_get_the_excerpt( $post = null, $allowed_html = null ) {
 			$excerpt = wp_trim_words( $excerpt, $excerpt_length, $excerpt_more );
 		}
 
-		// Remove all shortcode Content before removing HTML
-		if ( ! $allow_shortcode ) {
+		// If shortcode filter is enabled lets process them
+		if ( $allow_shortcode ) {
+			$excerpt = do_shortcode( $excerpt );
```

Comment:
If I'm remembering correctly - I may not be ;-) - the reason we added code to strip anything that even _looks_ like a shortcode is because during ajax requests we return the response so early it's not uncommon for some shortcodes to not have been registered.

In this approach, we'd still run that risk (ie, if `[3rd_party_magic]` won't be registered until after the point where we return our response, then it still won't be processed and we'll be left with what looks like an unparsed shortcode in the output).

What about this: instead of an either/or approach, we could default to always running `do_shortcode()` first and stripping unparsed shortcodes out second. That way shortcodes that have been registered early enough would still be rendered, but we'd clean up the detritus of shortcodes that weren't already registered (and we could still use filters/hooks to allow fine grained control over this policy).